homework/list.js

```diff
@@ -52,28 +43,48 @@
       buff.appendChild(li);
     });
     ul.appendChild(buff);
-    _wrapper.appendChild(ul);
-  }
+    this._wrapper.appendChild(ul);
+  },
 
-  function fetchList(afterFetch) {
-    var xhr = new XMLHttpRequest();
-    xhr.open('GET', 'http://127.0.0.1:8000/demo-list-notes.json', true);
-    xhr.responseType = 'json';
-    xhr.onreadystatechange = function(e) {
-      // Watch out: we have a mysterious unknown 'this'.
-      if (this.readyState === 4 && this.status === 200) {
-        var listData = this.response;
-        // The flow ends here.
-        afterFetch(listData);
-      } else if (this.status !== 200 ){
+  fetchList() {
+    return new Promise((function(resolve,reject) {
+      var xhr = new XMLHttpRequest();
+      xhr.open('GET', 'demo-list-notes.json', true);
+      xhr.responseType = 'json';
+      xhr.onreadystatechange = function(e) {
+        // Watch out: we have a mysterious unknown 'this'.
+        if (this.readyState === 4 && this.status === 200) {
+          var listData = this.response;
+          // The flow ends here.
+          resolve(listData);
+        } else if (this.status !== 200 ){
         // Ignore error in this case.
-      }
-    };
-    xhr.send();
-  }
+        }
+      };
+      xhr.send();
+    }).bind(this));
+  },
 
-  document.addEventListener('DOMContentLoaded', function(event) {
-    start();
-  });
+  handleEvent(event) {
+    switch(event.type) {
+      case 'click':
+        if(event.target.classList.contains('note-title')) {
+           this.onNoteOpen(event);
+        }
+      break;
+    }
+  },
+
+  start() {
+    window.addEventListener('click',this);
+    this._wrapper = document.querySelector('#note-list-wrapper');
+    this.fetchList().then((function(data) {
```

Comment:
Another point is in this way you didn't practice how to chain a Promise. So although it's unnecessary, please try to make a chain with several `then` and a `catch`.